app/uk/gov/hmrc/apiplatformevents/models/common/models.scala

```diff
@@ -55,7 +57,13 @@ object EventType extends  Enum[EventType] with PlayJsonEnum[EventType]  {
 
 }
 
+case class EventId(value: UUID) extends AnyVal
+object EventId {
+  def random: EventId = EventId(UUID.randomUUID())
+}
+
 trait ApplicationEvent{
+  val id: Option[EventId]
```

Comment:
I'll remove the option with the next change (the one to send the emails), where I'll also remove the scheduled job that populates the missing IDs.